src/fight/computation_tools.rs

```diff
@@ -0,0 +1,517 @@
+use crate::fight::global_values;
+use crate::{model, regiment};
+
+/// Compute the value to hit the opponent
+///
+/// # Parameters
+/// offensive (usize): The offensive Stats of the attacking Regiment
+///
+/// defense (usize): The defense Stats of the defending Regiment
+pub fn compute_roll_to_hit(offensive: usize, defense: usize) -> usize {
+    let difference: i8 = offensive as i8 - defense as i8;
+
+    match difference {
+        i8::MIN..=-8 => 6,
+        -7..=-4 => 5,
+        -3..=0 => 4,
+        1..=3 => 3,
+        4..=i8::MAX => 2,
+    }
+}
+
+/// Compute the value to wound the opponent
+///
+/// # Parameters
+/// strength (usize): The strength Stats of the attacking Regiment
+///
+/// resilience (usize): The resilience Stats of the defending Regiment
+pub fn compute_roll_to_wound(strength: usize, resilience: usize) -> usize {
+    let difference: i8 = strength as i8 - resilience as i8;
+
+    match difference {
+        i8::MIN..=-2 => 6,
+        -1 => 5,
+        0 => 4,
+        1 => 3,
+        2..=i8::MAX => 2,
+    }
+}
+
+/// Compute the number of attacks of a Regiment
+///
+/// # Parameters
+/// regiment (regiment::Regiment): The Regiment attacking
+///
+/// # Return
+/// usize: The number of attacks of the Regiment
+pub fn get_nb_attacks(regiment_attacking: &regiment::Regiment) -> usize {
+    (regiment_attacking.get_model().get_stats().get_attack() as f64
+        * regiment_attacking.get_cols() as f64
+        * 1.5)
+        .floor() as usize
+}
+
+/// ## Compute the probability for a model to wound an another
+///
+/// ### Parameters
+/// (&Unit) main -> The attacker
+///
+/// (&Unit) target -> The defender
+///
+/// ### Return
+/// isize -> The probability for main to wound target
+pub fn compute_wound_probability(attacking: &model::Model, defending: &model::Model) -> f64 {
+    let minimum_to_hit: usize = compute_roll_to_hit(
+        attacking.get_stats().get_offensive(),
+        defending.get_stats().get_defense(),
```

Comment:
We call get_stats twice for each regiment, by storing the return value of get_stat() we will avoid one call per regiment
I think its a loss of efficiency to create two variables to avoid only two calls to simple functions like getters

Reply:
Yes but if you change the parameter to get stats instead of the whole Model you can change it in `compute_save_probability`too and remove 4 calls to get_stats maybe it's better ?

Reply:
And still calling a function takes more times than getting a value from a registry